sdk/trace/src/main/java/io/opentelemetry/sdk/trace/export/BatchSpanProcessor.java

```diff
@@ -61,7 +61,11 @@
  *   <li>{@code OTEL_BSP_EXPORT_TIMEOUT}: sets the maximum allowed time to export data.
  *   <li>{@code OTEL_BSP_EXPORT_SAMPLED}: sets whether only sampled spans should be exported.
  * </ul>
+ *
+ * @deprecated Use {@link
+ *     io.opentelemetry.sdk.trace.SdkTracerProviderBuilder#addExporter(SpanExporter)}
  */
+@Deprecated
```

Comment:
I don't think we should deprecate the batch span processor, or remove it. This is something that is basically required by the current SDK specification and I would not remove it without some significant spec rewording.

Reply:
I think even with this approach we are satisfying everything here

https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/sdk.md#span-processor

- Built in processors can be registered (`BatchSettings`)

- Custom processors can be registered (`addSpanProcessor`)

- Built in processors can be decorated (`addSpanProcessor`)

I would rather expose BSP later if it becomes really necessary if we are ok with the approach in this PR. Or otherwise close the PR and live with registering an exporter being a bit tedious.